src/wallet/walletutil.h

```diff
@@ -86,4 +87,38 @@ class WalletLocation final
     bool Exists() const;
 };
 
+/** Descriptor with some wallet metadata */
+class WalletDescriptor
+{
+public:
+    std::shared_ptr<Descriptor> descriptor;
+    uint64_t creation_time;
+    int32_t range_start; // First item in range; start of range, inclusive, i.e. [range_start, range_end). This never changes.
+    int32_t range_end; // Item after the last; end of range, exclusive, i.e. [range_start, range_end). This will increment with each TopUp()
+    int32_t next_index; // Position of the next item to generate
+    DescriptorCache cache;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        if (ser_action.ForRead()) {
+            std::string desc;
+            std::string error;
+            READWRITE(desc);
+            FlatSigningProvider keys;
+            descriptor = Parse(desc, keys, error, true);
+        } else {
+            READWRITE(descriptor->ToString());
```

Comment:
Descriptors are inherently strings. There is no alternative serialization that doesn't eventually boil down to a string.

Reply:
It doesn't have to be that way. We describe "wrapped segwit script" with "sh(wsh(script))", but it might just as well have been "p2sh[wsh[script]]". There's already talk of tweaking descriptors to make them more compatible with miniscript. We use base58 encoded xpubs, but we might as well switch to something bech32 encoded in the future. It makes sense to have a more computer-friendly representation, that's mapped 1 to 1 with these strings. See also #18043.

It's just that I'd rather not delay this PR for that.

Reply:
We're not going to get this right on the first shot likely no matter what we decide. As long as it's not a supreme burden to upgrade I think current is fine.